sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Customizations/ContainerRegistryCredentials.cs

```diff
@@ -280,29 +281,31 @@ private string GetScopeFromHeaders(HttpHeaders headers)
                 if (headerKVP.Key.ToLower() == challengeHeader)
                 {
                     headerValue = string.Join(",", headerKVP.Value);
-                    
+
                     break;
                 }
             }
 
-            string scope = "";
             int position = headerValue.IndexOf("scope=");
-            if (position > 0)
+            string scope = headerValue.Substring(position);
+            string[] keyValues = scope.Split('=');
+            int length = keyValues.Length;
+            var scopeContainedIn = keyValues[1];
```

Comment:
At this point, the length hasn't been checked so you could get a `IndexOutOfRangeException`